src/key.rs

```diff
@@ -462,6 +480,10 @@ fn to_raw_mode(mode: KeyMode) -> raw::KeyMode {
     }
 }
 
+/// # Panics
+///
+/// Will panic if `RedisModule_KeyType` or `RedisModule_ModuleTypeGetType` are missing in redismodule.h
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
```

Comment:
Adding here `#[allow(clippy::not_unsafe_ptr_arg_deref)]` in order to to avoid adding `unsafe`, which then would also require adding `unsafe` to all callers recursively?
And since this would probably affect many callers and would require many unsafe blocks?
So we prefer to add this configuration and silence this error (`deny` level) since we know the low level APIs should exist?